include/exadg/convection_diffusion/user_interface/parameters.cpp

```diff
@@ -256,6 +257,21 @@ Parameters::check() const
   // SPATIAL DISCRETIZATION
   grid.check();
 
+  if(enable_adaptivity)
+  {
+    AssertThrow(not ale_formulation,
+                dealii::ExcMessage("Combination of adaptive mesh refinement "
+                                   "and ALE formulation not implemented."));
+
+    AssertThrow(temporal_discretization == TemporalDiscretization::BDF,
+                dealii::ExcMessage("Combination of adaptive mesh refinement "
+                                   "and explicit time integration not implemented."));
+
+    AssertThrow(use_cell_based_face_loops == false,
+                dealii::ExcMessage("Cell-based face loops not supported for "
+                                   "adaptively refined meshes."));
```

Comment:
I could imagine we can realize this assert more generically (using `triangulation.has_hanging_nodes()` rather than `enable_adaptivity`), so that we do not have to add such an assert for every module. If I remember correctly, we introduced this recently. See https://github.com/exadg/exadg/blob/6fd3bee0a4db8455e42ebc0e067ea6a2ad8cc66c/include/exadg/matrix_free/categorization.h#L41

Reply:
with PR #617, the respective assert in `do_cell_based_loops()` will be triggered as intended, so that we can then remove this assert on the level of the ConvDiff module.